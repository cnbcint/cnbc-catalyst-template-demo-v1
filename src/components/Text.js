import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Pdrop = styled.p`
  color: ${(props) => props.color};
  font-size: ${(props) => (props.highlight ? "1.4em" : "1.2rem")};
  line-height: ${(props) => (props.highlight ? "140%" : "125%")};
  text-align: ${(props) => props.textAlign};
  max-width: 600px;
  margin:0 auto;
  &::first-letter {
    color: ${(props) => props.dropCapColor};
    padding: 0 0;
    margin: 0.5rem 1rem 0 0;
    font-size: 5.5rem;
    font-weight: bold;
    float: left;
    line-height: 4rem;
  }
`;
const P = styled.p`
  color: ${(props) => props.color};
  font-size: ${(props) => (props.highlight ? "1.4rem" : "1.2rem")};
  line-height: ${(props) => (props.highlight ? "140%" : "125%")};
  text-align: ${(props) => props.textAlign};
  max-width: 600px;
  margin:0 auto;

`;
/** 
 Text Block will add a paragraph element to it's children. Do not add P inside this element to avoid double nested P element
*/
export default function Text(props) {
  if (props.dropCap) {
    return (
      <Pdrop
        highlight={props.highlight}
        dropCapColor={props.dropCapColor ? props.dropCapColor : props.color}
        color={props.color}
        textAlign={props.textAlign}
        style={props.style}
        className={props.className}
      >
        {props.children}
      </Pdrop>
    );
  }
  return (
    <P
      highlight={props.highlight}
      color={props.color}
      style={props.style}
      textAlign={props.textAlign}
      className={props.className}
    >
      {props.children}
    </P>
  );
}

Text.defaulProps = {
  dropCap: false,
  color: "inherit",
  dropCapColor: "",
  highlight: false,
  className: "",
  textAlign: "left",
};

Text.propTypes = {
  /**
   Will Increase font size to highlight the text
   */
  highlight: PropTypes.bool,
  /**
   Will Increase font size of the first letter of the paragraph to achive DropCap 
   */
  dropCap: PropTypes.bool,
  /**
   CSS override for font color
   */
  color: PropTypes.string,
  /**
   CSS text aligment
   */
  textAlign: PropTypes.oneOf(["left", "right", "center"]),
  /**
   CSS color for drop cap
   */
  dropCapColor: PropTypes.string,
  /**
  React CSS format to override rules has required by design
  */
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
};
