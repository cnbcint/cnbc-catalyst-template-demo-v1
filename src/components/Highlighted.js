import React, { useContext } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { LayoutContext } from "./Main";
import { getMargin } from "../helpers/Layout";
import useAnimation from "../hooks/useAnimation";

const Container = styled.div`
  font-weight: bold;
  font-size: 110%;
`;
const Divider = styled.hr`
  margin: ${getMargin(1)} 0;
  border-color: ${(props) => props.color};
`;

export default function Highlighted(props) {
  const layout = useContext(LayoutContext);
  const { ref: refDevider } = useAnimation({
    animationName: "fadeIn",
    duration: 500,
    delay: 125,
    loop: true,
    autoStart: true,
  });
  const { ref: refContent } = useAnimation({
    animationName: "fadeInLeft",
    duration: 500,
    delay: 0,
    loop: true,
    autoStart: true,
  });
  return (
    <Container className={props.className} style={props.style}>
      <div ref={refContent}>{props.children}</div>
      <Divider ref={refDevider} color={layout.colors[props.dividerColor]} />
    </Container>
  );
}

Highlighted.defaultProps = {
  dividerColor: "primary",
  className: "",
  style: {},
};
Highlighted.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  style: PropTypes.object,
  dividerColor: PropTypes.oneOf(["primary", "secondary", "tertiary"]),
};
