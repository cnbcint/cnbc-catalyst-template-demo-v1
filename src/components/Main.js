import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin, breakpoints } from "../helpers/Layout";
import { ModalContext } from "../components/Modal";

export const LayoutContext = React.createContext({
  maxWidth: "500px",
  colors: null,
});

const Container = styled.div`
  display: block;
  width: 100%;
  background-color: ${(props) => props.colors.background};
  > * {
    font-size: 1.125rem;
    line-height: 140%;
    color: ${(props) => props.colors.copy};
  }
  hr {
    // border-color: ${(props) => props.primaryColor};
    border-color: ${(props) => props.colors.copy};
    border-style: solid;
    display: inline-block;
    width: 100%;
    // margin-top: ${getMargin(1)};
    margin-bottom: ${getMargin(1)};
    @media (min-width: ${breakpoints.mobile}) {
      // margin-top: ${getMargin(2)};
      margin-bottom: ${getMargin(2)};
    }
  }
  img {
    width: 100%;
    display: block;
    margin-top: ${getMargin(2)};
    margin-bottom: ${getMargin(2)};
  }
  p {
    margin: 1rem auto !important;
  }
  ul {
    list-style: disc !important;
    margin-top: ${getMargin(1)} !important;
    margin-bottom: ${getMargin(1)} !important;
    li {
      list-style: disc !important;
      margin: ${getMargin(1)} !important;
      margin-left: ${getMargin(2)} !important;
    }
  }
`;
export default function Main(props) {
  return (
    <LayoutContext.Provider
      value={{ maxWidth: props.maxWidth, colors: props.colors }}
    >
      <ModalContext.Provider value={{ modals: {} }}>
        <Container
          colors={props.colors}
          primaryColor={props.colors.primary}
          className={props.className}
          style={props.style}
        >
          {props.children}
        </Container>
      </ModalContext.Provider>
    </LayoutContext.Provider>
  );
}
Main.defaultProps = {
  maxWidth: "900px",
  colors: { primary: "black", secondary: "gray", tertiary: "red" },
};
Main.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  maxWidth: PropTypes.string,
  colors: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string,
    background: PropTypes.string,
    copy: PropTypes.string,
  }),
  style: PropTypes.object,
};
