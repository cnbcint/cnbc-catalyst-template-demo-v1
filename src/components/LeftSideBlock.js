import React, { useContext } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { breakpoints, getMargin } from "../helpers/Layout";
import { LayoutContext } from "./Main";

const Container = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  box-sizing: border-box;
  align-items: center;
  margin-top: ${getMargin(3)};
  margin-bottom: ${getMargin(3)};
  @media (min-width: ${breakpoints.mobile}) {
    margin-top: ${getMargin(5)};
    margin-bottom: ${getMargin(5)};
  }
`;
const Side = styled.div`
  width: 80%;

  box-sizing: border-box;
  @media (min-width: ${breakpoints.mobile}) {
    flex-direction: row;
    min-width: ${(props) => props.sideWidth}px;
    width: ${(props) => props.sideWidth}px;
    padding-right: ${getMargin(2)};
  }
`;
const Content = styled.div`
  width: 100%;
  max-width: calc(${(props) => props.maxWidth});
  display: flex;
  flex-direction: column;
  flex-shrink: 1;
  @media (min-width: ${breakpoints.mobile}) {
    width: calc(100% - ${getMargin(2)});
  }
`;
const Wrapper = styled.div`
  max-width: calc(
    ${(props) => props.maxWidth} + ${(props) => 2 * props.sideWidth}px
  );
  display: flex;
  width: calc(100% - ${getMargin(2)});
  align-items: center;
  flex-direction: column;
  @media (min-width: ${breakpoints.mobile}) {
    flex-direction: row;
    align-items: baseline;
  }
`;

export default function LeftSideBlock(props) {
  const layout = useContext(LayoutContext);
  return (
    <Container className={props.className} style={props.style}>
      <Wrapper sideWidth={props.sideWidth} maxWidth={layout.maxWidth}>
        <Side
          sideWidth={props.sideWidth}
          className={props.className ? props.className + "__side" : ""}
        >
          {props.sideContent()}
        </Side>
        <Content
          className={props.className ? props.className + "__content" : ""}
          maxWidth={layout.maxWidth}
        >
          {props.children}{" "}
        </Content>
      </Wrapper>
    </Container>
  );
}
LeftSideBlock.defaultProps = {
  sideWidth: 300,
};

LeftSideBlock.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,

  /**
    React CSS format to override rules has required by design
    */
  style: PropTypes.object,

  /**
    The content that will be displayed on the left side
    */
  sideContent: PropTypes.func,

  /**
    The width of the left side content
    */
  sideWidth: PropTypes.number,
};
