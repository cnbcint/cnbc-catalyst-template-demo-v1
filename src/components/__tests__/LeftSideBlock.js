import React from "react";
import { render } from "@testing-library/react";
import "jest-styled-components";
import Main from "../Main";
import LeftSideBlock from "../LeftSideBlock";
import Text from "../Text";
import Highlighted from "../Highlighted";

const Theme = {
  maxWidth: "900px",
  colors: {
    primary: "orange",
    secondary: "green",
    tertiary: "orange",
    background: "white",
    copy: "black",
    links: "blue",
  },
};
const Component = () => (
  <Main {...Theme}>
    <LeftSideBlock
      style={{ color: "red" }}
      className="test"
      sideContent={() => <Highlighted>Testing side content</Highlighted>}
    >
      <Text>Testing content</Text>
    </LeftSideBlock>
  </Main>
);

beforeAll(() => {
  window.IntersectionObserver = class {
    observe() {
      return null;
    }
    disconnect() {}
  };
});

describe("LeftSideBlock", () => {
  it("Renders", () => {
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });

  it("Renders the side content", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test__side")[0];
    expect(target.firstChild.firstChild.textContent).toMatch(
      "Testing side content"
    );
  });

  it("Renders the correct content", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test__content")[0];
    expect(target.firstChild.textContent).toMatch("Testing content");
  });

  it("Passes the class name", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test")[0];
    expect(target.classList.contains("test")).toBe(true);
  });

  it("Passes the style", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test")[0];

    expect(target.style["color"]).toBe("red");
  });
});

afterAll(() => {
  window.IntersectionObserver = null;
});
