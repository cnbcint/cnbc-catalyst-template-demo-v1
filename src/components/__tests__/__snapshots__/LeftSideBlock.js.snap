// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftSideBlock Renders 1`] = `
.c0 {
  display: block;
  width: 100%;
  background-color: white;
}

.c0 > * {
  font-size: 1.125rem;
  line-height: 140%;
  color: black;
}

.c0 hr {
  border-color: orange;
  border-style: solid;
  display: inline-block;
  width: 100%;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c0 img {
  width: 100%;
  display: block;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul {
  list-style: disc !important;
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul li {
  list-style: disc !important;
  margin: 1rem !important;
  margin-left: 2rem !important;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.c3 {
  width: 80%;
  box-sizing: border-box;
}

.c6 {
  width: 100%;
  max-width: calc(900px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c2 {
  max-width: calc( 900px + 600px );
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 2rem);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  font-size: 1.2rem;
  line-height: 125%;
}

.c4 {
  font-weight: bold;
  font-size: 110%;
}

.c5 {
  margin: 1rem 0;
  border-color: orange;
}

@media (min-width:768px) {
  .c0 hr {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
}

@media (min-width:768px) {
  .c1 {
    margin-top: 5rem;
    margin-bottom: 5rem;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    min-width: 300px;
    width: 300px;
    padding-right: 2rem;
  }
}

@media (min-width:768px) {
  .c6 {
    width: calc(100% - 2rem);
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: baseline;
    -webkit-box-align: baseline;
    -ms-flex-align: baseline;
    align-items: baseline;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 test"
      style="color: red;"
    >
      <div
        class="c2"
      >
        <div
          class="c3 test__side"
        >
          <div
            class="c4"
          >
            <div
              class="pre-animated"
            >
              Testing side content
            </div>
            <hr
              class="c5 pre-animated"
              color="orange"
            />
          </div>
        </div>
        <div
          class="c6 test__content"
        >
          <p
            class="c7"
          >
            Testing content
          </p>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
